### PR TITLE
Issue #14759: Updated Command Line Options in table format

### LINKED ISSUE #14759: Enhance Command Line documentation by organizing options in Tabular format
Extension of PR #14710 and response to https://github.com/checkstyle/checkstyle/pull/14710#discussion_r1550049366

Current cmdline docs : https://checkstyle.org/cmdline.html#Command_line_usage
Our command line options are bulleted under one section at present.

A better way of documenting for end users would be organizing all the available cmdline options in a Tabular format.
The table would consist of 2 columns (Titles can be "`Command Line Option`" and "`Description`" or decided further) 
and 18 rows for each of the 18 options that we host.

This format of documentation is the standard and can be found followed by Apache maven CLI options too
https://maven.apache.org/ref/3.6.1/maven-embedder/cli.html#

### PR DESCRIPTION
**Pull request by kkoutsilis**
Closes #14759 
Reused code from #14799, and added anchor image  (as per https://github.com/checkstyle/checkstyle/pull/14799#issuecomment-2356627003)

### CODE REVIEW

**Comment by romani** [comment]
GitHub, generate website

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java [hunk 1]
```diff
@@ -63,24 +59,24 @@ public void validateCliDocSections() throws Exception {
...
-            cmdDesc.remove(option);
+                    .that(descMain)
+                    .isEqualTo(descXdoc);
+            cmdOptions.remove(option);
         }
         assertWithMessage("CLI Options: %s present in cmdline.xml.vm, not documented in Main.java",
-                    cmdDesc)
-                .that(cmdDesc)
+                cmdOptions.values())
+                .that(cmdOptions.values())
```
Comment (kkoutsilis) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:79]: Actually this should be either
```suggestion
                cmdOptions)
                .that(cmdOptions)
```
or
```suggestion
                cmdOptions.keys())
                .that(cmdOptions.keys())
```
No real preference

Comment (rdiachenko) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:79]: Let's use the first option 'that(cmdOptions)'

Comment (kkoutsilis) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java]: Updated

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java]: @kkoutsilis , please do not resolve conversations.
Please allow reviewer to confirm that all is done as required and threre is no misunderstndings.

**Review by romani — APPROVED** [review]
ok to merge.

thanks a lot for help!!

**Comment by romani** [comment]
@kkoutsilis , as you PRs becomes a bit more complicated from logic perspective, review  will takes longer.
Please do not hesitate to send next PRs, no need to wait for this  PR to be merged. Handling few opened PR is completely ok.

**Comment by kkoutsilis** [comment]
> @kkoutsilis , as you PRs becomes a bit more complicated from logic perspective, review will takes longer. Please do not hesitate to send next PRs, no need to wait for this PR to be merged. Handling few opened PR is completely ok.

During the week I have limited time due to work, so not sure if I will be able to work on multiple issues, but will keep that in mind. Thanks @romani

**Comment by romani** [comment]
That is one of reasons we will be slow too

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**